src/SquidNew.cc

```diff
@@ -22,6 +22,10 @@ void operator delete(void *address)
 {
     xfree(address);
 }
+void operator delete(void* address, std::size_t)
+{
+    xfree(address);
+}
```

Comment:
Why are we adding this C++14 delete [operator](https://en.cppreference.com/w/cpp/memory/new/operator_delete)? We are still on C++11.

N.B. The whole idea of customizing _some_ of the new/delete operators on _some_ compilers feels problematic to me. I do not know whether this PR can deal with this problem though. This paragraph does not negate the previous paragraph -- we should address the C++14 problem in the PR that introduces it, of course.

Reply:
Because otherwise:

SquidNew.cc:21:6: error: the program should also define ‘void operator delete(void*, std::size_t)’ [-Werror=sized-deallocation]
   21 | void operator delete(void *address)
      |      ^~~~~~~~

kinkie@nuc:~/src/squid$ gcc --version
gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.

Reply:
I agree with you on the "some new/delete operators on some compilers" statement. That's a problem for another PR tho.

Reply:
> SquidNew.cc:21:6: error: the program should also define ‘void operator delete(void*, std::size_t)’ [-Werror=sized-deallocation]

This is a C++14 error. According to my [test](https://godbolt.org/z/doYeKaKKd), the error is detected only if C++14 support is enabled. AFAICT, C++14 is enabled in GCC _by default_ since GCC v6.

By adding a C++14 declaration, we are risking the "opposite" error in compilers that do not yet support C++14. However, based on my quick tests (which is not the best way to establish such truths, of course), C++ may allow custom delete operators with virtually any additional parameters, so the added delete operator may be harmless in those older compilers -- it will be treated as a custom operator that the old compiler will never use on its own.

Let's clarify what is going on and avoid code duplication:

```suggestion
// Squid does not use C++14 yet, but this declaration avoids a
// -Wsized-deallocation error when building with C++14 implicitly enabled.
void operator delete(void* address, std::size_t)
{
    operator delete(address);
}
```

----

This brings an interesting question: Should we force C++11 support (to exclude support for higher standards)? I think we should when we build with `-Werror`. However, I do not insist on the corresponding compiler flags computation change, primarily because I think we should get rid of `-Werror` instead.

Reply:
I would also recommend `#including` one of the required standard headers that [declare](https://en.cppreference.com/w/cpp/types/size_t) `std::size_t` which was not present in this compilation unit before. I bet the code "works" without that because we are including one of those headers indirectly, but I am not sure that is an invariant we should rely on.

Alternatively, use `size_t` instead of `std::size` in the added delete operator. The `size_t` type has the same `#include` problem, but this file is already using `size_t`, so we would not be adding any new problems and do not have to solve the old ones.

Reply:
We are testing this against a number of compilers all the way back to gcc-4.8.5 and it seems to be working well.
Interestingly, I can't find -std=c++11 in the build logs. This is consistent with the behaviour in trunk tho, so no unwanted changes happened

Reply:
> I can't find -std=c++11 in the build logs

According to (my interpretation of) its documentation, AX_CXX_COMPILE_STDCXX_11() does not add `-std=c++11` when that flag is not needed to enable C++11 support.